detectable/src/main/java/com/blackduck/integration/detectable/detectables/gradle/inspection/parse/GradleReportLineParser.java

```diff
@@ -38,17 +41,22 @@ public class GradleReportLineParser {
     private final Map<String, Map<String, String>> transitiveRichVersions = new HashMap<>();
 
     // This map is handling all the child-parent relationships which are found in the entire project
-    // with each child as a key and their respective parent as the value.
-    private final Map<String, String> relationsMap = new HashMap<>();
+    // with each child as a key and their parents (if more than one) in a HashSet .
```

Comment:
My understanding is that there can only be one parent.
Is it more like a set of **possible** parents? And if so, are we guessing which one applies? And if so, does it mean we might sometimes guess incorrectly and get an incorrect version?
I wonder if there is a more precise to key projects with something in addition to name and level.

Reply:
I understand your concern and in order to make sure, I have added project path from the metadata we generate to correct this. As an example, if sub declares rich version and foo is child of both sub and subtwo, we change version if :sub:foo is the path we are parsing and do not change if we are parsing :subtwo:foo. I have also confirmed for various projects, this project path is generated from settings.gradle so it is a reliable source.